src/main/java/io/wisetime/connector/jira/database/JiraDb.java

```diff
@@ -4,27 +4,38 @@
 
 package io.wisetime.connector.jira.database;
 
+import com.google.common.annotations.VisibleForTesting;
 import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
 import com.google.inject.Inject;
 
+import org.apache.commons.lang3.tuple.Pair;
 import org.codejargon.fluentjdbc.api.FluentJdbcException;
 import org.codejargon.fluentjdbc.api.mapper.Mappers;
 import org.codejargon.fluentjdbc.api.query.Query;
 
 import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
 import java.util.List;
 import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
 
+import io.wisetime.connector.jira.models.ImmutableIssue;
 import io.wisetime.connector.jira.models.Issue;
 import io.wisetime.connector.jira.models.Worklog;
 
 /**
  * Simple, unsophisticated access to the Jira database.
  *
  * @author shane.xie@practiceinsight.io
+ * @author alvin.llobrera@practiceinsight.io
  */
 public class JiraDb {
 
+  private final DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
```

Comment:
Looks like timezone isn't saved in the worklog created date column. Do you know if Jira expects the datetime to be in UTC?

Reply:
Jira expects the datetime to be in the Jira's default time zone which can be configured during installation. By default it is using the operating system's timezone.
https://confluence.atlassian.com/kb/how-to-set-the-timezone-for-the-java-environment-841187402.html

I was able to confirm it by changing my profile to a different timezone and after creating worklog via Jira UI, it saves the datetime in the Jira default time zone.

One improvement we could have is to auto detect Jira default time zone because Jira allows the default time zone to be modified, see https://community.atlassian.com/t5/Jira-questions/Is-it-possible-to-edit-default-timezone-for-JIRA/qaq-p/7270.
It would be nice if our Jira connector will automatically detect that change. Thoughts?

Reply:
Yes, let's auto-detect the correct timezone to save the time under, and get rid of our configuration option. When time is sent from WT, the segment hour, it is always in UTC. We'll need to then convert to Jira's configured timezone.

Reply:
Ok will take a look